backend/src/crd/controller/scheduledworkflow/main.go

```diff
@@ -97,28 +119,24 @@ func main() {
 	}
 }
 
-func initEnv() {
+func init() {
+
 	// Import environment variable, support nested vars e.g. OBJECTSTORECONFIG_ACCESSKEY
 	replacer := strings.NewReplacer(".", "_")
 	viper.SetEnvKeyReplacer(replacer)
 	viper.AutomaticEnv()
 	viper.AllowEmptyEnv(true)
-}
-
-func init() {
-	initEnv()
 
 	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&namespace, "namespace", "", "The namespace name used for Kubernetes informers to obtain the listers.")
+	flag.StringVar(&mlPipelineAPIServerName, "pipelineAPIServerName", "ml-pipeline", "Name of the ML pipeline API server.")
+	flag.StringVar(&mlPipelineAPIServerBasePath, "pipelineAPIServerBasePath",
+		"/apis/v1beta1", "The base path for the ML pipeline API server.")
+	flag.StringVar(&mlPipelineServiceHttpPort, "pipelineServiceHttpPort", "8888", "Http Port of the ML pipeline API server.")
+	flag.StringVar(&mlPipelineServiceGRPCPort, "pipelineServiceGRPCPort", "8887", "GRPC Port of the ML pipeline API server.")
```

Comment:
will we be able to get these values from env vars like `ML_PIPELINE_SERVICE_HOST` and `ML_PIPELINE_SERVICE_PORT`?

Reply:
How would `ML_PIPELINE_SERVICE_HOST ` replace `mlPipelineAPIServerBasePath ` and `mlPipelineAPIServerName` I might be missing something here?

Reply:
I see, since we already have the util, let's keep using it.